### PR TITLE
Add support for .desk file

### PR DESCRIPTION
**Pull request by ianberinger**
Thanks for sharing your project!
This PR adds support for activating a desk based on the content of a `.desk` file in the current directory (issue #42).
I updated the documentation accordingly.
Feel free to close if you feel this doesn't fit the spirit of the project :)

### CODE REVIEW

**Comment by jamesob** [comment]
Very cool, @ianberinger, and good start. Going to be taking a detailed look at this today or tomorrow.

**Comment by rnewton** [comment]
+1

**Comment by mblarsen** [comment]
+1 Just what I need

File: desk [hunk 1]
```diff
@@ -91,6 +92,11 @@ cmd_init() {
 
 cmd_go() {
     local TODESK="$1"
+    if [ -z "$TODESK" -a -f "${PWD}/.desk" ]; then
+	# if no desk is specified try the .desk file in the current directory
+	read TODESK < "${PWD}/.desk"
```
Comment (jamesob) [inline desk]: I'd prefer we go with `Deskfile` here (resembles `Makefile`, `Dockerfile`, etc.).

File: README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ There are no dependencies other than `bash`. Desk is explicitly tested with `bas
 `zsh`, and `fish`.
 
 ```sh
-◲  desk 0.3.2
+◲  desk 0.4.0
```
Comment (jamesob) [inline README.md]: This is probably the source of a merge conflict.

**Comment by jamesob** [comment]
Awesome. I'm +1 on this aside from a minor change in naming.

**Comment by ianberinger** [comment]
Thanks for reviewing. I renamed .desk to DESKFILE and fixed the merge conflict.

**Comment by rnewton** [comment]
I'm fine with .desk or Deskfile. DESKFILE is really loud.

**Comment by ianberinger** [comment]
Sorry, my bad. Don't know how I got to the all caps version. Fixed now.

**Comment by wricardo** [comment]
Really looking forward to this.

File: desk [hunk 3]
```diff
@@ -27,6 +27,7 @@ Usage:
         List all desks along with a description.
     $PROGRAM (.|go) <desk-name> [shell-args...]
         Activate a desk. Extra arguments are passed onto shell.
+        Without a name specified, it tries activating the desk specified in the ./.desk file.
```
Comment (jamesob) [inline desk]: out of date `.desk` reference

File: desk [hunk 4]
```diff
@@ -91,6 +92,11 @@ cmd_init() {
 
 cmd_go() {
     local TODESK="$1"
+    if [ -z "$TODESK" -a -f "${PWD}/Deskfile" ]; then
+	     # if no desk is specified try the Deskfile file in the current directory
```
Comment (jamesob) [inline desk]: Comment over-indented

**Comment by jamesob** [comment]
Feel bad blocking on small comments... I'm ready to merge this thing and iterate, though. I'll hold off on cutting a release after merge because I want to add tests and spruce up the documentation. This is close, folks! Thanks for all the work @ianberinger @rnewton!

**Comment by jamesob** [comment]
Unfortunately this doesn't actually work for me and I've had to revert: 
```sh
(venv)  [Sat 13 18:41] job/code/desk master* ±
 $ ls Deskfile

Deskfile

(venv)  [Sat 13 18:42] job/code/desk master* ±
 $ desk .

Desk # Description: the desk I use to work on desk :) (# Description: the desk I use to work on desk :).sh) not found in /Users/job/.desk/desks
```

I'll revisit this feature tomorrow when I have some time to code.

**Comment by ianberinger** [comment]
@jamesob, I didn't account for comments in the Deskfile. I fixed that now:
https://github.com/ianberinger/desk/commit/e54c77eafc216c50390eb7c1efb3274e603798ca ; line 97
Should I open another PR

**Pull request merged**